js/index.js

```diff
@@ -1,42 +1,92 @@
 // ITERATION 1
 
 function updateSubtotal(product) {
-  console.log('Calculating subtotal, yey!');
+  
+  
 
   //... your code goes here
+  const price = product.querySelector('.price span');
+  const quantity = product.querySelector('.quantity input');
+  const subtotal = product.querySelector('.subtotal span');
+
+
+  subtotal.innerHTML = price.innerHTML * quantity.value;
+  
+  
 }
 
 function calculateAll() {
   // code in the following two lines is added just for testing purposes.
   // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
-
-  // ITERATION 2
-  //... your code goes here
-
-  // ITERATION 3
-  //... your code goes here
+  const singleProduct = document.querySelectorAll('.product');
+  const allSubTotal = document.querySelectorAll('.subtotal span');
+  let total = 0;
+  let result = document.querySelector('#total-value span');
+  
+  for (let i = 0; i < singleProduct.length; i++) {
+    updateSubtotal(singleProduct[i]);
+    
+    let num = parseInt(allSubTotal[i].innerHTML);
```

Comment:
You could also return the subtotal price from the function "updateSubtotal" instead of getting the value from the document again